tests/Integration/Behaviour/Features/Context/Domain/Product/SpecificPricePrioritiesFeatureContext.php

```diff
@@ -0,0 +1,100 @@
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain\Product;
+
+use Behat\Gherkin\Node\TableNode;
+use PHPUnit\Framework\Assert;
+use PrestaShop\PrestaShop\Core\Domain\Exception\DomainException;
+use PrestaShop\PrestaShop\Core\Domain\SpecificPrice\Command\SetGlobalSpecificPricePriorityCommand;
+use PrestaShop\PrestaShop\Core\Domain\SpecificPrice\Command\SetSpecificPricePriorityForProductCommand;
+use SpecificPrice;
+use Tests\Integration\Behaviour\Features\Context\Util\PrimitiveUtils;
+
+class SpecificPricePrioritiesFeatureContext extends AbstractProductFeatureContext
+{
+    /**
+     * @When I set following specific price priorities for product :productReference:
+     *
+     * @param string $productReference
+     * @param TableNode $prioritiesTable
+     */
+    public function setPrioritiesForSingleProduct(string $productReference, TableNode $prioritiesTable): void
+    {
+        $priorities = PrimitiveUtils::castStringArrayIntoArray($prioritiesTable->getRow(0)[0]);
+
+        try {
+            SpecificPrice::flushCache();
+            $this->getCommandBus()->handle(new SetSpecificPricePriorityForProductCommand(
+                $this->getSharedStorage()->get($productReference),
+                $priorities
+            ));
+        } catch (DomainException $e) {
+            $this->setLastException($e);
+        }
+    }
+
+    /**
+     * @When I set following specific price priorities for all products:
+     *
+     * @param TableNode $prioritiesTable
+     */
+    public function setGlobalPriorities(TableNode $prioritiesTable): void
+    {
+        $priorities = PrimitiveUtils::castStringArrayIntoArray($prioritiesTable->getRow(0)[0]);
+
+        try {
+            SpecificPrice::flushCache();
+            $this->getCommandBus()->handle(new SetGlobalSpecificPricePriorityCommand($priorities));
+        } catch (DomainException $e) {
+            $this->setLastException($e);
+        }
+    }
+
+    /**
+     * @Then product :productReference should have following specific price priorities:
+     *
+     * @param string $productReference
+     * @param TableNode $prioritiesTable
+     */
+    public function assertProductPriorities(string $productReference, TableNode $prioritiesTable): void
+    {
+        $expectedPriorities = PrimitiveUtils::castStringArrayIntoArray($prioritiesTable->getRow(0)[0]);
```

Comment:
The formatting and access of behat values i a bit weird, especially the way it's written with a formatted array inside a table..

Maybe it would be more readable this way (to keep the one-line format):

```
    When I set following specific price priorities for product "product1":
      | id_country | id_currency | id_group | id_shop |
    Then product "product1" should have following specific price priorities:
      | id_country | id_currency | id_group | id_shop |
```

Which should be accessible with `$prioritiesTable->getRow(0)` I think 🤔  (or close)

Reply:
yep that should be better